pkg/controller/daemon/daemon_controller.go

```diff
@@ -684,12 +684,8 @@ func nodeInSameCondition(old []v1.NodeCondition, cur []v1.NodeCondition) bool {
 }
 
 func shouldIgnoreNodeUpdate(oldNode, curNode v1.Node) bool {
-	if !nodeInSameCondition(oldNode.Status.Conditions, curNode.Status.Conditions) {
-		return false
-	}
-	oldNode.ResourceVersion = curNode.ResourceVersion
-	oldNode.Status.Conditions = curNode.Status.Conditions
-	return apiequality.Semantic.DeepEqual(oldNode, curNode)
+	return apiequality.Semantic.DeepEqual(oldNode.GetLabels(), curNode.GetLabels()) &&
```

Comment:
```suggestion
	return apiequality.Semantic.DeepEqual(oldNode.GetLabels, curNode.GetLabels) &&
```

Reply:
If modified to `return apiequality.Semantic.DeepEqual(oldNode.GetLabels, curNode.GetLabels)` when Node labels and taints have not changed, shouldIgnoreNodeUpdate will return false. Not as expected.

Reply:
I guess it was supposed to be: `oldNode.Labels, curNode.Labels`

Reply:
> I guess it was supposed to be: `oldNode.Labels, curNode.Labels`

Yes, that's right.